scripts/ci/ci_pylint.sh

```diff
@@ -17,14 +17,53 @@
 # limitations under the License.
 #
 
-# Script to run Pylint on all code. Should be started from root directory of Airflow:
+# Script to run Pylint on all code. Can be started from any working directory
 # ./scripts/ci/ci_pylint.sh
 
-set -ex
+set -uo pipefail
+
+# Uncomment to see the commands executed
+# set -x
+
+MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+pushd ${MY_DIR}/../../ || exit 1
+
+echo
+echo "Running in $(pwd)"
+echo
+
+echo
+echo "Running pylint for source code without tests"
+echo
 
 find . -name "*.py" \
 -not -path "./.eggs/*" \
 -not -path "./airflow/www/node_modules/*" \
 -not -path "./airflow/_vendor/*" \
 -not -path "./build/*" \
+-not -path "./tests/*" \
```

Comment:
`git ls-files airflow/ | grep '.py'` ?

Reply:
I like this, but again I have concerns that in some cases we might not have .git around. Do you think it is worth it to rely on .git dir @ashb

Reply:
I can't see a case we'd want to run Pylint without a checkout as it's very much a CI-only thing. (I'm guessing we will volume mount in a checkout under the docker-based CI you will be working on?)

Reply:
Here is the case explained: https://github.com/GoogleCloudPlatform/cloud-builders/issues/236 - the more "modern" CIs (such as cloud buid) work exclusively as Docker-first and usually the will ignore .git repo because otherwise it has to be passed as context to within Docker. That might slow-down the process a lot for big repositories as context is often sent to remote Docker  server. 

I am not saying we are going to do this, but I think building in a dependency on .git might not be future-proof as we know some CIs are already going in this direction.

Reply:
This is exactly what happens in Dockerhub, so this is not an isolated cloud-build case only.

Reply:
I use airflow-breeze to run Airflow, but I do small modification in my environment. I use a [git worktree](https://git-scm.com/docs/git-worktree] to eliminate the need to have multiple copies of the same repository in multiple workspace.  In this case, the git does not work properly. I would like to run pylint in this environment. Not relying on git is a very good idea.